src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/DSACryptoServiceProvider.cs

```diff
@@ -0,0 +1,506 @@
+            }
+            set
+            {
+                bool oldPersistKeyInCsp = this.PersistKeyInCsp;
+                if (value == oldPersistKeyInCsp)
+                {
+                    return; // Do nothing
+                }
+                CapiHelper.SetPersistKeyInCsp(SafeProvHandle, value);
+            }
+        }
+
+        /// <summary>
+        /// Gets the information of key if it is a public key
+        /// </summary>
+        public bool PublicOnly
+        {
+            get
+            {
+                byte[] publicKey = CapiHelper.GetKeyParameter(SafeKeyHandle, Constants.CLR_PUBLICKEYONLY);
+                return (publicKey[0] == 1);
+            }
+        }
+
+        /// <summary>
+        /// MachineKey store properties
+        /// </summary>
+        public static bool UseMachineKeyStore
+        {
+            get
+            {
+                return (s_useMachineKeyStore == CspProviderFlags.UseMachineKeyStore);
+            }
+            set
+            {
+                s_useMachineKeyStore = (value ? CspProviderFlags.UseMachineKeyStore : 0);
+            }
+        }
+
+        /// <summary>
+        /// Dispose the key handles 
+        /// </summary>
+        protected override void Dispose(bool disposing)
+        {
+            base.Dispose(disposing);
+
+            if (_safeKeyHandle != null && !_safeKeyHandle.IsClosed)
+            {
+                _safeKeyHandle.Dispose();
+            }
+
+            if (_safeProvHandle != null && !_safeProvHandle.IsClosed)
+            {
+                _safeProvHandle.Dispose();
+            }
+        }
+
+        /// <summary>
+        /// Exports a blob containing the key information associated with an DSACryptoServiceProvider object.
+        /// </summary>
+        public byte[] ExportCspBlob(bool includePrivateParameters)
+        {
+            return CapiHelper.ExportKeyBlob(includePrivateParameters, SafeKeyHandle);
+        }
+
+        /// <summary>
+        /// Exports the DSAParameters
+        /// </summary>
+        public override DSAParameters ExportParameters(bool includePrivateParameters)
+        {
+            byte[] cspBlob = ExportCspBlob(includePrivateParameters);
+            return cspBlob.ToDSAParameters(includePrivateParameters, SafeKeyHandle);
+        }
+
+        /// <summary>
+        /// This method helps Acquire the default CSP and avoids the need for static SafeProvHandle
+        /// in CapiHelper class
+        /// </summary>
+        private SafeProvHandle AcquireSafeProviderHandle()
+        {
+            SafeProvHandle safeProvHandle;
+            CapiHelper.AcquireCsp(new CspParameters(CapiHelper.DefaultDssProviderType), out safeProvHandle);
+            return safeProvHandle;
+        }
+
+        /// <summary>
+        /// Imports a blob that represents DSA key information
+        /// </summary>
+        /// <param name="keyBlob"></param>
```

Comment:
Actually not sure what we should be doing here regarding best practice. Most originated from RSA side (which were superfluous), but I changed some of the 'important' ones to match msdn. Now I plan to remove the comments if they are overrides of base class or simple properties (redundant wording) and adding any missing msdn comments.